.github/workflows/tests.yaml

```diff
@@ -65,3 +67,29 @@ jobs:
       - name: Run tests
         run: |
           go test -v -race -cover -short ./...
+  linux-32bit:
+    name: "unit tests (linux/386)"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Go 1.18
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Run tests
+        run: |
+          GOARCH=386 go test -v -cover -short ./...
```

Comment:
I think we can use `env` for this: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idenv

I think that also means we can combine the linux and windows jobs into a single job with a matrix between the two as well